### PR TITLE
MAINT: Port read-joining tutorial

### LINKED ISSUE #269: Port read-joining community tutorial

### PR DESCRIPTION
**Pull request by thermokarst**
Fixes #269 

---

If we decide to keep the links to the outstanding issues in this tutorial (see below), when we merge this, lets update those issues with reminders to update this tutorial when those changes are made.

### CODE REVIEW

File: source/tutorials/read-joining.rst [hunk 1]
```diff
@@ -0,0 +1,223 @@
+Alternative methods of read-joining in QIIME 2
+==============================================
+
+.. note:: This tutorial does not cover read-joining and denoising with
+   DADA2. Instead, this tutorial focuses on alternative approaches to analyzing
+   paired end reads in QIIME 2. If you are interested in joining and denoising
```
Comment (nbokulich) [inline source/tutorials/read-joining.rst]: paired end --> paired-end

File: source/tutorials/read-joining.rst [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+   reads with DADA2, the :doc:`Atacama soil microbiome tutorial
+   <atacama-soils>` illustrates how to use ``qiime dada2 denoise-paired`` for
+   this purpose.  If you plan to use DADA2 to join and denoising your paired
```
Comment (nbokulich) [inline source/tutorials/read-joining.rst]: denoising --> denoise

File: source/tutorials/read-joining.rst [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+   end data, **do not join your reads prior to denoising with DADA2;** DADA2
+   expects reads that have not yet been joined, and will join the reads for you
+   during the denoising process.
+
+In QIIME 2, we use the term *single end reads* to refer to forward or reverse
```
Comment (nbokulich) [inline source/tutorials/read-joining.rst]: single end --> single-end
paired end --> paired-end

File: source/tutorials/read-joining.rst [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+reads in isolation; we use the term *paired end reads* to refer to forward and
+reverse reads that have not yet been joined; and we use the term *joined reads*
+to refer to forward and reverse reads that have already been joined (or
+merged). **It is important to understand which of these terms apply to your
+data, as this will determine what steps are necessary to analyze your paired
```
Comment (nbokulich) [inline source/tutorials/read-joining.rst]: paired end --> paired-end

File: source/tutorials/read-joining.rst [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+end data.**
+
+It is currently possible to join paired end reads in QIIME 2 using the
```
Comment (nbokulich) [inline source/tutorials/read-joining.rst]: paired-end

File: source/tutorials/read-joining.rst [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+
+.. command-block::
+
+   qiime quality-filter q-score-joined \
+     --i-demux demux-joined.qza \
+     --o-filtered-sequences demux-joined-filtered.qza \
+     --o-filter-stats demux-joined-filter-stats.qza
+
+At this stage you can choose to proceed using `Deblur`_ for additional quality
+control, or you can dereplicate sequences and optionally cluster them into OTUs
```
Comment (nbokulich) [inline source/tutorials/read-joining.rst]: provide a link? or perhaps you are subtly discouraging irresponsible use of vsearch... friends don't let friends pick otus? 🍻 🚗 💥

File: source/tutorials/read-joining.rst [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+with ``q2-vsearch``. Deblur should give much higher quality results, so we
+recommend that procedure and will illustrate that approach in the next steps of
+this tutorial.
+
+If you are instead interested in experimenting with an analysis workflow that
+is more like QIIME 1 processing (for example, to compare your Deblur or DADA2
+result with a QIIME 1-like pipeline), you should next dereplicate and cluster
+your sequences. If you try this option, we strongly encourage you to call
+``qiime quality-filter q-score-joined`` with a higher ``min-quality`` threshold
+- possibly ``--p-min-quality 20`` or ``--p-min-quality 30``. You can then
```
Comment (nbokulich) [inline source/tutorials/read-joining.rst]: maybe link to [my work on this](https://doi.org/10.1038/nmeth.2276)?

File: source/tutorials/read-joining.rst [hunk 8]
```diff
@@ -0,0 +1,223 @@
...
+follow the steps in the `OTU clustering tutorial`_. After clustering, you will
+likely want to filter features that are observed in only one sample using
+``qiime feature-table filter-features --p-min-samples``.
```
Comment (nbokulich) [inline source/tutorials/read-joining.rst]: setting an abundance filter is also advisable (see my ancient work linked above). At least filter out singletons!

Chimera filtering is also strongly advised with OTU picking.

File: source/tutorials/read-joining.rst [hunk 9]
```diff
@@ -0,0 +1,223 @@
...
+In QIIME 2, we use the term *single end reads* to refer to forward or reverse
+reads in isolation; we use the term *paired end reads* to refer to forward and
+reverse reads that have not yet been joined; and we use the term *joined reads*
+to refer to forward and reverse reads that have already been joined (or
+merged). **It is important to understand which of these terms apply to your
+data, as this will determine what steps are necessary to analyze your paired
+end data.**
+
+It is currently possible to join paired end reads in QIIME 2 using the
+``q2-vsearch`` plugin, or to import reads that have been joined outside of
```
Comment (nbokulich) [inline source/tutorials/read-joining.rst]: perhaps include a link to the section below on importing joined reads?

**Comment by thermokarst** [comment]
Thanks @nbokulich! Just pushed up some changes!

File: source/tutorials/read-joining.rst [hunk 10]
```diff
@@ -3,26 +3,26 @@ Alternative methods of read-joining in QIIME 2
 
 .. note:: This tutorial does not cover read-joining and denoising with
    DADA2. Instead, this tutorial focuses on alternative approaches to analyzing
-   paired end reads in QIIME 2. If you are interested in joining and denoising
+   paired-end reads in QIIME 2. If you are interested in joining and denoising
    reads with DADA2, the :doc:`Atacama soil microbiome tutorial
    <atacama-soils>` illustrates how to use ``qiime dada2 denoise-paired`` for
-   this purpose.  If you plan to use DADA2 to join and denoising your paired
-   end data, **do not join your reads prior to denoising with DADA2;** DADA2
+   this purpose.  If you plan to use DADA2 to join and denois your paired end
```
Comment (nbokulich) [inline source/tutorials/read-joining.rst]: denois --> denoise

File: source/tutorials/read-joining.rst [hunk 11]
```diff
@@ -3,26 +3,26 @@ Alternative methods of read-joining in QIIME 2
...
 reverse reads that have not yet been joined; and we use the term *joined reads*
 to refer to forward and reverse reads that have already been joined (or
 merged). **It is important to understand which of these terms apply to your
-data, as this will determine what steps are necessary to analyze your paired
-end data.**
+data, as this will determine what steps are necessary to analyze your
+paired-end data.**
 
-It is currently possible to join paired end reads in QIIME 2 using the
+It is currently possible to join paired end-reads in QIIME 2 using the
```
Comment (nbokulich) [inline source/tutorials/read-joining.rst]: paired end-reads --> paired-end reads

**Comment by thermokarst** [comment]
*nervous laugh*

Thanks @nbokulich!

**Comment by nbokulich** [comment]
looks good, thanks @thermokarst !

**Pull request merged**